integration-tests/tests/src/tests/schema.ts

```diff
@@ -44,6 +44,32 @@ describe("Realm schema", () => {
         return realm.create<Test>("Test", {});
       });
 
+      expect(test.dynamic).to.equal(42);
+    });
+    it("can take a function as a default property value for a primary key", () => {
+      interface Test {
+        dynamic?: number;
+      }
+
+      const realm = new Realm({
+        schema: [
+          {
+            name: "Test",
+            primaryKey: "dynamic",
+            properties: {
+              dynamic: {
+                type: "int",
+                default: () => 42,
+              },
+            },
+          },
+        ],
+      });
+
+      const test = realm.write(() => {
+        return realm.create<Test>("Test", {});
```

Comment:
This won't fail your test (although it should).

```suggestion
        return realm.create<Test>("Test", { dynamic: 0 });
```

Reply:
Why not?  it doesn't equal 42

Reply:
![image](https://user-images.githubusercontent.com/164606/222735173-f6f4b94c-e2e3-441d-a1e1-19903ed1f6c3.png)
With your change it fails

Reply:
Thats unrelated (and because you're missing a call to `Realm.clearTestState`).

Reply:
What I wanted to show was that a falsy value would be replaced by the default.

Reply:
I ran the single test, but good call in adding `Realm.clearTestState`

Reply:
Ah...you are right.  I added a test to check this as well with the change you provided.  Falsey values now work as well.